test/functional/interface_rpc.py

```diff
@@ -91,6 +122,17 @@ def test_http_status_codes(self):
         assert_equal(response['error']['code'], -32700)                 # RPC_PARSE_ERROR
         assert_equal(status, 500)
 
+        self.log.info("Testing HTTP status codes for JSON-RPC 2.0 requests...")
+        AuthServiceProxy.jsonrpc20 = True
+        # All 200!
+        expect_http_status(200, -32601, self.nodes[0].invalidmethod)    # RPC_METHOD_NOT_FOUND
+        expect_http_status(200, -8,     self.nodes[0].getblockhash, 42) # RPC_INVALID_PARAMETER
+        expect_http_status(200, None,   self.nodes[0].getblockhash, 0)
+        # force-send invalidly formatted request
+        response, status = self.nodes[0]._request('POST', "/", b'this is bad')
+        assert_equal(response['error']['code'], -32700)                 # RPC_PARSE_ERROR
+        assert_equal(status, 500)
```

Comment:
This error is caused by a failure to parse JSON in `ReadBody()` so theoretically we also can't read the `"jsonrpc": "2.0"` pair either. I agree with you though, the HTTP transport is fine and should be 200, because the error is coming from the json parser. I guess the options are:
1. always send 200 [here](https://github.com/bitcoin/bitcoin/blob/master/src/httprpc.cpp#L77) and kinda break backwards compat
2. attempt to parse the `"jsonrpc"` string and only send 200 if we can detect the "2.0" value
3. leave as-is, meaning invalid requests will get 500 even if they otherwise intended to be "2.0"